lib/text/lrc_text_parser.js

```diff
@@ -0,0 +1,121 @@
+/*! @license
+ * Shaka Player
+ * Copyright 2016 Google LLC
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+goog.provide('shaka.text.LrcTextParser');
+
+goog.require('goog.asserts');
+goog.require('shaka.log');
+goog.require('shaka.text.Cue');
+goog.require('shaka.text.TextEngine');
+goog.require('shaka.util.StringUtils');
+
+
+/**
+ * @implements {shaka.extern.TextParser}
+ * @export
+ */
+shaka.text.LrcTextParser = class {
+  /**
+   * @override
+   * @export
+   */
+  parseInit(data) {
+    goog.asserts.assert(false, 'LRC does not have init segments');
+  }
+
+  /**
+   * @override
+   * @export
+   */
+  parseMedia(data, time) {
+    return shaka.text.LrcTextParser.getCues(data);
+  }
+
+  /**
+   * @param {BufferSource} data
+   * @return {!Array.<!shaka.extern.Cue>}
+   * @export
+   */
+  static getCues(data) {
+    const StringUtils = shaka.util.StringUtils;
+    const LrcTextParser = shaka.text.LrcTextParser;
+
+    // Get the input as a string.
+    const str = StringUtils.fromUTF8(data);
+
+    /** @type {shaka.extern.Cue} */
+    let prevCue = null;
+
+    /** @type {!Array.<!shaka.extern.Cue>} */
+    const cues = [];
+    const parts = str.split(/\r?\n/);
```

Comment:
These parts are really lines of the input, so I think "lines" would be a clearer name.  I initially assumed these were parts of a single line.

Reply:
Done!